refinedstorage2-platform-forge/src/main/java/com/refinedmods/refinedstorage2/platform/forge/integration/recipemod/rei/GridIngredientConverter.java

```diff
@@ -0,0 +1,27 @@
+package com.refinedmods.refinedstorage2.platform.forge.integration.recipemod.rei;
+
+import com.refinedmods.refinedstorage2.platform.api.integration.recipemod.IngredientConverter;
+import com.refinedmods.refinedstorage2.platform.common.internal.grid.view.FluidGridResource;
+import com.refinedmods.refinedstorage2.platform.common.internal.grid.view.ItemGridResource;
+
+import java.util.Optional;
+
+import me.shedaniel.rei.api.common.util.EntryStacks;
+
+public class GridIngredientConverter implements IngredientConverter {
+    @Override
+    public Optional<Object> convertToResource(final Object ingredient) {
+        return Optional.empty();
+    }
+
+    @Override
+    public Optional<Object> convertToIngredient(final Object resource) {
+        if (resource instanceof ItemGridResource itemResource) {
+            return Optional.of(EntryStacks.of(itemResource.copyItemStack()));
+        }
+        if (resource instanceof FluidGridResource fluidResource) {
+            return Optional.of(EntryStacks.of(fluidResource.copyFluidResource().fluid()));
```

Comment:
how would I get the nbt on the fluid for rei to recognize it?

Reply:
works for forge packages, not for fabric, since some classnames are mangled. As discussed in discord, that should wait for fabric until we can get back info from our dependency rei here.